src/core/logging/logger.hpp

```diff
@@ -324,14 +329,16 @@
 
 #endif
 
-#define log_and_throw(message)                                      \
-  do {                                                              \
-    auto throw_error = [&]() GL_COLD_NOINLINE_ERROR {  \
-      logstream(LOG_ERROR) << (message) << std::endl;               \
-      throw(std::string(message));                                  \
-    };                                                              \
-    throw_error();                                                  \
-  } while(0)
+#ifdef NDEBUG
+
+#define log_and_throw(message)                        \
+  do {                                                \
+    auto throw_error = [&]() GL_COLD_NOINLINE_ERROR { \
+      logstream(LOG_ERROR) << (message) << std::endl; \
+      throw(std::runtime_error(message);              \
```

Comment:
Will this change how errors are handled in Python or through the C API `tc_error *` mechanism?

Reply:
Let me check how error is handled in C API.

Reply:
This should be fine.